### ISSUE TITLE
color NCS 0580-G76Y is broken

### ISSUE THREAD

**Issue by milahu** [m90/ncs-color#5]
of the many colors, [NCS 0580-G76Y](https://www.w3schools.com/colors/colors_converter.asp?color=ncs(0580-G76Y)) is broken:

```
272/400 deg -> NCS 0580-G72Y -> #f7ff23 hex = rgb(247,255,35) rgb
273/400 deg -> NCS 0580-G73Y -> #f9ff24 hex = rgb(249,255,36) rgb
274/400 deg -> NCS 0580-G74Y -> #fbff24 hex = rgb(251,255,36) rgb
275/400 deg -> NCS 0580-G75Y -> #fdff24 hex = rgb(253,255,36) rgb
276/400 deg -> NCS 0580-G76Y -> #16516533 hex = rgb(357,357,51) rgb
277/400 deg -> NCS 0580-G77Y -> #fefd24 hex = rgb(254,253,36) rgb
278/400 deg -> NCS 0580-G78Y -> #fffb24 hex = rgb(255,251,36) rgb
279/400 deg -> NCS 0580-G79Y -> #fff924 hex = rgb(255,249,36) rgb
```

it should be similar to its neighbors, but the hex and rgb values are out of range

value should be `rgb(255, 255, 51)` according to `w3color.js`

**Comment by m90** [comment]
So I debugged this a little by dumping the local variables for a run and it seems that `NCS 0580-G76Y` fails on extracting the "blackness" correctly.

This is what I see

```
NCS 0580-G75Y

Ba: 0
Bc: 0.11701717492464797
C: 80
C1: "G"
Cn: 80
Ga: 0.88125
Gc: 0.822017174924648
N: 75
Ra: 0.8740076238697193
Rc: 0.8162232740204234
S: 0
Sn: 5
ncs: (6) ["NCS 0580-G75Y", "05", "80", "G", "75", "Y", index: 0, input: "NCS 0580-G75Y", groups: undefined]
ss: 1.2165195941212144
this: global
top: 0.822017174924648
v: undefined
x1: -95
x2: 0.5850858746232398
x3: undefined
x5: undefined
x6: undefined
x7: 15
x8: undefined
```
```
NCS 0580-G76Y

Ba: 0
Bc: 0.11733333333333335
C: 80
C1: "G"
Cn: 80
Ga: 0.88
Gc: 0.8213333333333334
N: 76
Ra: 0.88
Rc: 0.8213333333333334
S: 0
Sn: 5
ncs: (6) ["NCS 0580-G76Y", "05", "80", "G", "76", "Y", index: 0, input: "NCS 0580-G76Y", groups: undefined]
ss: 1.7045454545454546
this: global
top: 0.5866666666666667
v: undefined
x1: -94
x2: 0.5866666666666667
x3: undefined
x5: undefined
x6: undefined
x7: 16
x8: undefined
```
```
NCS 0580-G77Y

Ba: 0
Bc: 0.11764377871566867
C: 80
C1: "G"
Cn: 80
Ga: 0.87875
Gc: 0.8206437787156687
N: 77
Ra: 0.88590668073503
Rc: 0.8263691233036926
S: 0
Sn: 5
ncs: (6) ["NCS 0580-G77Y", "05", "80", "G", "77", "Y", index: 0, input: "NCS 0580-G77Y", groups: undefined]
ss: 1.210112977118698
this: global
top: 0.8263691233036926
v: undefined
x1: -93
x2: 0.5882188935783433
x3: undefined
x5: undefined
x6: undefined
x7: 17
x8: undefined
```

where you can see that the `top` value is an outlier for the failing color.

It looks like the problem comes from this part of the code where the failing case unexpectedly hits the last branch:

https://github.com/m90/ncs-color/blob/57ae2b27c670346498341137e3d536f3eb301dba/ncscolor.js#L112-L121

I don't have the time to dive deeper into this right now, but maybe it gives you a starting point in case you're interested.

**Comment by m90** [comment]
The problem is probably `Rc` and `Gc` being the exact same in the failing case, which is not catered for by the check.

**Comment by milahu** [comment]
yepp:

```
ncscolor.js:129 Rc 0.8162232740204234 + Gc 0.822017174924648 + Bc 0.11701717492464797
ncscolor.js:136 top = Gc
App.svelte:616 275/400 deg -> NCS 0580-G75Y -> #fdff24 hex = rgb(253,255,36) rgb

ncscolor.js:129 Rc 0.8213333333333334 + Gc 0.8213333333333334 + Bc 0.11733333333333335
ncscolor.js:142 top = (Rc + Gc + Bc) / 3
App.svelte:616 276/400 deg -> NCS 0580-G76Y -> #16516533 hex = rgb(357,357,51) rgb

ncscolor.js:129 Rc 0.8263691233036926 + Gc 0.8206437787156687 + Bc 0.11764377871566867
ncscolor.js:133 top = Rc
App.svelte:616 277/400 deg -> NCS 0580-G77Y -> #fefd24 hex = rgb(254,253,36) rgb
```

fixed in #6
